sql/core/src/test/scala/org/apache/spark/sql/tpcdsDagagen.scala

```diff
@@ -0,0 +1,443 @@
+          args = tail
+
+        case "--tableFilter" :: value :: tail =>
+          tableFilter = value
+          args = tail
+
+        case "--numPartitions" :: value :: tail =>
+          numPartitions = toPositiveIntValue("Number of partitions", value)
+          args = tail
+
+        case "--help" :: tail =>
+          printUsageAndExit(0)
+
+        case _ =>
+          // scalastyle:off println
+          System.err.println("Unknown/unsupported param " + args)
+          // scalastyle:on println
+          printUsageAndExit(1)
+      }
+    }
+
+    checkRequiredArguments()
+  }
+
+  private def printUsageAndExit(exitCode: Int): Unit = {
+    // scalastyle:off
+    System.err.println("""
+      |build/sbt "test:runMain <this class> [Options]"
+      |Options:
+      |  --master                        the Spark master to use, default to local[*]
+      |  --dsdgenDir                     location of dsdgen
+      |  --location                      root directory of location to generate data in
+      |  --scaleFactor                   size of the dataset to generate (in GB)
+      |  --format                        generated data format, Parquet, ORC ...
+      |  --overwrite                     whether to overwrite the data that is already there
+      |  --partitionTables               whether to create the partitioned fact tables
+      |  --clusterByPartitionColumns     whether to shuffle to get partitions coalesced into single files
+      |  --filterOutNullPartitionValues  whether to filter out the partition with NULL key value
+      |  --tableFilter                   comma-separated list of table names to generate (e.g., store_sales,store_returns),
+      |                                  all the tables are generated by default
+      |  --numPartitions                 how many dsdgen partitions to run - number of input tasks
+      """.stripMargin)
+    // scalastyle:on
+    System.exit(exitCode)
+  }
+
+  private def toPositiveIntValue(name: String, v: String): Int = {
+    if (Try(v.toInt).getOrElse(-1) <= 0) {
+      // scalastyle:off println
+      System.err.println(s"$name must be a positive number")
+      // scalastyle:on println
+      printUsageAndExit(-1)
+    }
+    v.toInt
+  }
+
+  private def checkRequiredArguments(): Unit = {
+    if (dsdgenDir == null) {
+      // scalastyle:off println
+      System.err.println("Must specify a dsdgen path")
+      // scalastyle:on println
+      printUsageAndExit(-1)
+    }
+    if (location == null) {
+      // scalastyle:off println
+      System.err.println("Must specify an output location")
+      // scalastyle:on println
+      printUsageAndExit(-1)
+    }
+  }
+}
+
+/**
+ * This class generates TPCDS table data by using tpcds-kit:
+ *  - https://github.com/databricks/tpcds-kit
+ *
+ * To run this:
+ * {{{
+ *   build/sbt "sql/test:runMain <this class> --dsdgenDir <path> --location <path> --scaleFactor 1"
+ * }}}
+ */
+object GenTPCDSData {
```

Comment:
Shall we rename the file `tpcdsDatagen.scala` -> `GenTPCDSData.scala`?

Reply:
sure